spock-specs/src/test/groovy/org/spockframework/smoke/WithBlockFailingConditions.groovy

```diff
@@ -155,4 +156,146 @@ class WithBlockFailingConditions extends Specification {
       size() == 3
     }
   }
+
+  @FailsWith(ConditionNotSatisfiedError)
+  def "GDK method that looks like built-in method as implicit condition"() {
+    expect:
+    null.with {
+      false
+    }
+  }
+
+  @FailsWith(ConditionNotSatisfiedError)
+  def "GDK method that looks like built-in method as explicit condition"() {
+    expect:
+    assert null.with {
+      false
+    }
```

Comment:
🤔 Do we need to test `assert` in other places (blocks, helper methods, ...)?

Reply:
Generally it could make sense for each and every test to have an implicit and an explicit version I guess. :-D

But thanks to Groovy adapting an earlier version of our power assertions, you should even get nice output and failure if the rewriting of Spock does not happen in most cases.

Specifically here was the problem that there was a half rewrite.
The rewriting removed the explicit `assert`, but did not add the `verifyMethodCondition` magic.
So we lost the Groovy built-in assertion and also the Spock assertion in this case.